### PR TITLE
Add zellkonverter to wilms-06 environment

### PR DESCRIPTION
**Pull request by sjspielman**
Towards #866 

We need to update the link to the CELLxGENE kidney reference used in this module to the h5ad AnnData file rather than the rds Seurat file, since CELLxGENE will stop supporting Seurat objects at the end of 2024. 

For this, we'll need the `zellkonverter` package. This PR updates the `renv` environment so that the Docker image can be updated, which will allow for a 2nd PR that downloads and converts the h5ad file. This PR also restores the R version to 4.4.0, the official OpenScPCA R version, from 4.4.1 which ended up in there at one point. Finally, I also updated the docker GHA to run on `feature/*` branches so will actually accomplish what we want it to do!

### CODE REVIEW

**Review by jaclyn-taroni — APPROVED** [review]
👍🏻 if checks pass 🏁

**Comment by sjspielman** [comment]
I need to move `zellkonverter` into `components/dependencies.R` for now!

**Comment by jashapiro** [comment]
For the docker image, you probably want to add a RUN step like the one in `doublet-detection` and elsewhere that pre-downloads the zellkonverter conda environment: https://github.com/AlexsLemonade/OpenScPCA-analysis/blob/ad8b1dc5e99716e515a74110053389f9d3b21bc3/analyses/doublet-detection/Dockerfile#L71-L75

File: analyses/cell-type-wilms-tumor-06/Dockerfile [hunk 1]
```diff
@@ -29,3 +29,9 @@ RUN Rscript -e 'renv::restore()' && \
   rm -rf ~/.cache/R/renv && \
   rm -rf /tmp/downloaded_packages && \
   rm -rf /tmp/Rtmp*
+
+# Complete installation of zellkonverter conda env
+ENV BASILISK_EXTERNAL_DIR /usr/local/renv/basilisk
+RUN Rscript -e "proc <- basilisk::basiliskStart(env = zellkonverter::zellkonverterAnnDataEnv(), testload = 'anndata'); \
+  basilisk::basiliskStop(proc); \
+  basilisk.utils::cleanConda()"
```
Comment (sjspielman) [inline analyses/cell-type-wilms-tumor-06/Dockerfile:37]:
```suggestion
  basilisk.utils::cleanConda()"

```

**Comment by sjspielman** [comment]
This has passed CI. The only change since was adding a newline for GitHub appeasement, so we can merge now.

**Pull request merged**